gravitee-apim-e2e/api-test/apis/api-publishing.spec.ts

```diff
@@ -35,18 +50,98 @@ describe('API - Publishing', () => {
         envId,
         newApiEntity: ApisFaker.newApi(),
       });
-      return userApi;
     });
 
     afterAll(async () => {
-      return await apisResourceUser.deleteApi({ orgId, envId, api: userApi.id });
+      await apisResourceUser.deleteApi({ orgId, envId, api: userApi.id });
     });
 
-    describe('As ANONYMOUS user', () => {
-      it('Get APIs should not contain created api', async () => {
+    describe.each`
+      user             | apiResource
+      ${'ANONYMOUS'}   | ${apiResourceAnonymous}
+      ${'ADMIN'}       | ${apiResourceAdmin}
+      ${'API_USER'}    | ${apiResourceApiUser}
+      ${'APP_USER'}    | ${apiResourceAppUser}
+      ${'SIMPLE_USER'} | ${apiResourceSimpleUser}
+    `('As $user user', ({ apiResource }) => {
+      test('Get APIs should not contain created api', async () => {
         const apisResponse = await apiResource.getApis({});
         expect(apisResponse.data.find((api) => api.id === userApi.id)).not.toBeDefined();
       });
+
+      test('Get API by id should return not found', async () => {
+        await fail(
+          apiResource.getApiByApiIdRaw({
```

Comment:
Minor remark for this `fail` call and others in this file :
We could call directly the `getApiByApiId` method instead of the `raw` one.